File: Projects/learn-spring-framework-01/src/main/java/com/feeeeel/learnspringframework/App03GamingSpringBeans.java
package com.feeeeel.learnspringframework;

import com.feeeeel.learnspringframework.game.GameRunner;
import com.feeeeel.learnspringframework.game.GamingConsole;
import com.feeeeel.learnspringframework.game.PacmanGame;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

public class App03GamingSpringBeans {
    public static void main(String[] args) {

        try (var context = new AnnotationConfigApplicationContext(GamingConfiguration.class)) {
            context.getBean(GamingConsole.class).up();

            context.getBean(GameRunner.class).run();
        }
    }
}

File: Projects/learn-spring-framwork-01-kotlin/src/main/kotlin/com/feeeeel/learnspringframwork01kotlin/App03GamingSpringBeans.kt
package com.feeeeel.learnspringframwork01kotlin

import com.feeeeel.learnspringframwork01kotlin.game.GameRunner
import com.feeeeel.learnspringframwork01kotlin.game.GamingConsole
import org.springframework.context.annotation.AnnotationConfigApplicationContext
import org.springframework.context.annotation.Configuration

class App03GamingSpringBeans {
}

fun main(args: Array<String>) {
    AnnotationConfigApplicationContext(GamingConfiguration::class.java).use { context ->
        context.getBean(GamingConsole::class.java).up()
        context.getBean(GameRunner::class.java).run()
    }
}